test/00_operators/OperatorTests.cu

```diff
@@ -1846,6 +1846,33 @@ TYPED_TEST(OperatorTestsNumericAllExecs, OperatorFuncs)
   res = c * c * (c + c) / c + three;
   EXPECT_TRUE(MatXUtils::MatXTypeCompare(tov0(), res, 0.07));
 
+  // example-begin nan-test-1
+  auto nan = make_tensor<TestType>({});
+  using conversionType = typename matx::detail::value_promote_t<TestType>;  
+  if constexpr(matx::is_complex_v<TestType>) {    
+    nan() = TestType(std::numeric_limits<conversionType>::quiet_NaN());
+  } else {
+    nan() = std::numeric_limits<conversionType>::quiet_NaN();
+  }
+  auto tob = make_tensor<bool>({});
+  (tob = matx::isnan(nan)).run(); 
+  cudaDeviceSynchronize();  
+  EXPECT_TRUE(MatXUtils::MatXTypeCompare(tob(), std::is_floating_point_v<conversionType> ? true : false));
+  // example-end nan-test-1
+
+  // example-begin inf-test-1
+  auto inf = make_tensor<TestType>({});
+  using conversionType = typename matx::detail::value_promote_t<TestType>;  
+  if constexpr(matx::is_complex_v<TestType>) {    
+    inf() = TestType(std::numeric_limits<conversionType>::infinity());
+  } else {
+    inf() = std::numeric_limits<conversionType>::infinity();
+  }  
+  (tob = matx::isinf(inf)).run(); 
+  cudaDeviceSynchronize();  
+  EXPECT_TRUE(MatXUtils::MatXTypeCompare(tob(), std::is_floating_point_v<conversionType> ? true : false));
```

Comment:
Maybe add a negative test case where you test numbers that are not NaN from random()?